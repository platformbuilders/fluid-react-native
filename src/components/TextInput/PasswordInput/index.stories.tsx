import React from 'react';
import { View } from 'react-native';
import type { Meta, StoryObj } from '@storybook/react';
import PasswordInput from '.';

const Render = ({ ...args }) => {
  return (
    <View style={{ width: 320 }}>
      <PasswordInput
        {...args}
        id="Password"
        accessibility="Este componente PasswordInput"
      />
    </View>
  );
};

const meta: Meta<typeof PasswordInput> = {
  title: 'PasswordInput',
  component: Render,
  args: {
    id: 'PasswordInput',
    label: 'Default',
    value: '',
  },
  parameters: {
    notes: 'Este componente Ã© utilizado para renderizar um PasswordInput.',
  },
};

export default meta;
type Story = StoryObj<typeof PasswordInput>;

export const Default: Story = {
  args: {
    suppressAnimation: false,
  },
};

export const Floating: Story = {
  args: {
    isFloating: true,
  },
};
