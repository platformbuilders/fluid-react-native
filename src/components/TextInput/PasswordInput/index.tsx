import React, { useState, FC, useCallback } from 'react';
import { TextInputType } from '../../../types';
import TextInput from '../index';

const PasswordInput: FC<TextInputType> = (props) => {
  const [hidePassword, setHidePassword] = useState<boolean>(true);
  const hitSlop = {
    left: 40,
    right: 40,
    top: 40,
    bottom: 40,
  };

  const onPressShowPassword = useCallback((): void => {
    setHidePassword(!hidePassword);
  }, [hidePassword]);

  return (
    <TextInput
      secureTextEntry={hidePassword}
      iconName={hidePassword ? 'eye' : 'eye-off'}
      iconTouchableEnabled
      onPressIcon={onPressShowPassword}
      iconHitSlop={hitSlop}
      {...props}
    />
  );
};

export default PasswordInput;
