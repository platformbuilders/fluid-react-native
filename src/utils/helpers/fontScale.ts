import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
