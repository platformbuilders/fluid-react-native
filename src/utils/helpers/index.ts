export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
