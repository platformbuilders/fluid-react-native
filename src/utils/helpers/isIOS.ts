import { Platform } from 'react-native';

const IOS_PLATFORM = 'ios';

export const isIOS = (): boolean => Platform.OS === IOS_PLATFORM;
