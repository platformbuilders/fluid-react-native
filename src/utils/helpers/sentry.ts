import { Sentry } from '.';

/**
 * Generic function to be used with Sentry. Under the hood, it's called the `addBreadcrumb` method
 * from Sentry API, which adds logs about user's actions prior to an error or crash.
 *
 * Read [this research doc](https://s3.us-west-2.amazonaws.com/secure.notion-static.com/89ba7171-90e3-4e5b-9856-fe36be0bed14/Sentry_-_Sentry_API_study_%28addBreadcrumb%29.pdf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=AKIAT73L2G45EIPT3X45%2F20230119%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20230119T180359Z&X-Amz-Expires=86400&X-Amz-Signature=b0395d3285cebaa0562ac64b30ed50a3803928181b7a40428c81dfd000442f03&X-Amz-SignedHeaders=host&response-content-disposition=filename%3D%22Sentry_-_Sentry_API_study_%28addBreadcrumb%29.pdf%22&x-id=GetObject)
 * to get more information about it's usage.
 * @param param0
 * @returns
 */
export const handleSentryTraceability = ({
  useSentryTraceability = false,
  category,
  data,
  event_id,
  level = 'info',
  message,
  timestamp,
  type,
}: HandleSentryTraceabilityProps) => {
  if (!useSentryTraceability) return;

  Sentry.addBreadcrumb({
    category,
    data,
    event_id,
    level,
    message,
    timestamp,
    type,
  });
};
