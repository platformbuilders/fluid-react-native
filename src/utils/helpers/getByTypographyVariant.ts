import { getTheme } from '@platformbuilders/helpers';
import { scaledFontSize } from './fontScale';

export const getFontSize = (props: any): number => {
  const fontSize = getTheme(`${props.variant}.fontSize`)(props) || 16;
  return scaledFontSize(fontSize);
};

export const getLineHeight = (props: any): number => {
  const lineHeight = getTheme(`${props.variant}.lineHeight`)(props) || 22;
  return scaledFontSize(lineHeight);
};
