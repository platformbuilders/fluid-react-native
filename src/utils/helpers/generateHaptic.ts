import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
