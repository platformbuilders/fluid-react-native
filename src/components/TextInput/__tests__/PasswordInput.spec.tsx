import 'jest';
import React, { useState, FC } from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';
import { ThemeProvider } from 'styled-components/native';
import PasswordInput from '../PasswordInput';
import { theme } from '../../../test/helpers';

describe('<PasswordInput />', () => {
  it('should render PasswordInput', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <PasswordInput id="test" accessibility="test" />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render input and trigger state change, blur, focus and onPressIcon', () => {
    const onBlurFN = jest.fn();
    const onFocusFN = jest.fn();
    const Component: FC = (): JSX.Element => {
      const [value, setValue] = useState('');

      return (
        <ThemeProvider theme={theme}>
          <PasswordInput
            id="testing_passwordInput"
            accessibility=""
            value={value}
            onChangeText={setValue}
            onBlur={onBlurFN}
            onFocus={onFocusFN}
          />
        </ThemeProvider>
      );
    };

    const { getByTestId } = render(<Component />);

    const component = getByTestId('testing_passwordInput');

    fireEvent.changeText(component, 'Value changed');
    fireEvent(component, 'blur');
    fireEvent(component, 'focus');

    expect(onBlurFN).toHaveBeenCalled();
    expect(onFocusFN).toHaveBeenCalled();
    expect(component.props.value).toBe('Value changed');
  });
});
