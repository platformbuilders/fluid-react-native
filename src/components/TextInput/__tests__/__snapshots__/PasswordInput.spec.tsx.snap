// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordInput /> should render PasswordInput 1`] = `
<View
  multiline={false}
  style={
    Array [
      Object {
        "justifyContent": "flex-start",
        "paddingTop": 0,
        "position": "relative",
      },
      Object {},
    ]
  }
>
  <Text
    accessibilityLabel="Erro test"
    contrast={false}
    status="DEFAULT"
    style={
      Array [
        Object {
          "color": "#212121",
          "fontSize": 31,
          "lineHeight": 43,
          "position": "absolute",
          "top": 20,
        },
        Object {
          "fontSize": 35,
          "top": 20,
        },
      ]
    }
    testID="error_test"
    variant="body"
  >
    
  </Text>
  <View
    multiline={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingTop": 18.857142857142858,
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      accessibility="test"
      accessibilityLabel="test"
      allowFontScaling={false}
      withoutBottomline={true}
      centered={false}
      contrast={false}
      id="test"
      isPlaceholder={true}
      keyboardType="default"
      large={false}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#72727260"
      rejectResponderTermination={true}
      secureTextEntry={true}
      status="DEFAULT"
      style={
        Array [
          Object {
            "borderWidth": 0,
            "color": "#212121",
            "flexGrow": 1,
            "fontSize": 33,
            "lineHeight": 43,
            "marginTop": 0,
            "minHeight": 37.714285714285715,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          Object {},
        ]
      }
      testID="test"
      textAlign="left"
      underlineColorAndroid="transparent"
      value=""
      variant="headline"
    />
    <View
      style={Object {}}
    >
      <View
        accessibilityLabel="icon_test"
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 40,
            "left": 40,
            "right": 40,
            "top": 40,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="id_eye"
      >
        <View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#C6B09E",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ó°ˆˆ
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    contrast={false}
    status="DEFAULT"
    style={
      Array [
        Object {
          "backgroundColor": "#212121",
          "height": 0.5,
        },
      ]
    }
  />
</View>
`;
